buildSrc/src/main/java/org/elasticsearch/gradle/test/ESTest.java

```diff
@@ -0,0 +1,47 @@
+package org.elasticsearch.gradle.test;
+
+import groovy.lang.Closure;
+import org.gradle.api.tasks.testing.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public class ESTest extends Test {
```

Comment:
I don't see the need for this. The argument provider exposed via `nonInputSystemProperties` extension does the same thing and supports lazy system properties already via lazy `GString`. Not to mention extending core task types, and overriding behavior is not generally recommended. If a lazy string is not sufficient and we need to be able to pass a `Supplier` or other `Callable` then I suggest we just add that logic [here](https://github.com/elastic/elasticsearch/blob/d8f4c150d6e0fdf176fa9ddfeef90aa133135b0f/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/RestIntegTestTask.groovy#L100-L102) rather than introduce a new subclass of `Test`.

Reply:
The main motivation was that the argument provider used to be tied to RestIntegTestTask when this was developed and I needed it on other test tasks. 
I would much prefer a `Supplier` or `Callable` as well as a more concise DSL for this, but there's no longer a need to do anything about it in this PR.

Reply:
Yeah, we attach that extension to all `Test` tasks now so that's unnecessary.